k9mail/src/main/java/com/fsck/k9/activity/misc/ContactPictureLoader.java

```diff
@@ -271,4 +332,24 @@ public void cancel() {
         }
     }
 
+    private Uri getGravatarUri(String email) {
+        return Uri.parse("http://www.gravatar.com/avatar/" + getMD5String(email) + "?d=404");
```

Comment:
`"http://www.gravatar.com/avatar/%s?d=404"` should be a constant format string defined somewhere reasonable (possibly `K9.GRAVATAR_URI_FORMAT` or in the XML).